src/utils.rs

```diff
@@ -134,6 +134,7 @@ pub(crate) fn random_plusminus_by_size_with_minimum<R: RngCore + CryptoRng>(
     use crate::errors::InternalError;
 
     if min >= max {
+        tracing::error!("Invalid arguments to random_plusminus_by_size_with_minimum (min >= max)");
```

Comment:
suggestion: the function name will get printed in the logs already, so I'd prefer a more expository error message that prints the bad values, like
> Can't sample from specified range because lower bound is not smaller than upper bound. \nLower bound: {}\nUpper bound: {}